### PR TITLE
Fix price slider implementation stripping parameters

### PR DESCRIPTION
**Pull request by Hlavtox**
| Questions         | Answers
| ------------------| -------------------------------------------------------
| Description?      | Price slider implementation was stripping all other parameters except `q`. For example, this resulted in breaking filtering on a search page.
| Type?             | 
| BC breaks?        | 
| Deprecations?     | 
| Fixed ticket?     | 
| Sponsor company   | 
| How to test?      |

### CODE REVIEW

File: src/js/modules/facetedsearch/filter-handler.ts [hunk 1]
```diff
@@ -3,31 +3,59 @@
...
 export default function (values: Array<string | number>, slider: API) {
   const {prestashop, Theme: {events}} = window;
 
-  const label = slider.target.dataset.sliderLabel;
-  const unit = slider.target.dataset.sliderUnit;
-  const encodedUrl = window.location.href;
-  const splittedUrl = encodedUrl.split('?');
-  let newUrl: string;
+  // Prepare query parameters
+  let queryParams = <any[]> [];
```
Comment (zuk3975) [inline src/js/modules/facetedsearch/filter-handler.ts:12]: I think this way should work
```suggestion
  let queryParams = Record<string, any> [];
```

File: src/js/modules/facetedsearch/filter-handler.ts [hunk 2]
```diff
@@ -3,31 +3,59 @@
...
 export default function (values: Array<string | number>, slider: API) {
   const {prestashop, Theme: {events}} = window;
 
-  const label = slider.target.dataset.sliderLabel;
-  const unit = slider.target.dataset.sliderUnit;
-  const encodedUrl = window.location.href;
-  const splittedUrl = encodedUrl.split('?');
-  let newUrl: string;
+  // Prepare query parameters
+  let queryParams = Record<string, any> [];
```
Comment (JBLach) [inline src/js/modules/facetedsearch/filter-handler.ts:12]:
```suggestion
  let queryParams = <any> [];
```

File: src/js/modules/facetedsearch/filter-handler.ts [hunk 3]
```diff
@@ -3,31 +3,59 @@
...
 
-    newUrl = `${splittedUrl[0]}?q=${groups.join('/')}`;
-  } else {
-    newUrl = `${splittedUrl[0]}?q=${label}-${unit}-${values[0]}-${values[1]}`;
+  if (!found) {
+    queryParams.push({name: 'q', value: ''});
   }
 
+  // Update query parameter
+  queryParams.forEach((query) => {
```
Comment (JBLach) [inline src/js/modules/facetedsearch/filter-handler.ts:38]:
```suggestion
  queryParams.forEach((query: any) => {
```

File: src/js/modules/facetedsearch/filter-handler.ts [hunk 4]
```diff
@@ -3,31 +3,59 @@
...
-      groups = groups.filter((e) => e.replace(label, '') === e);
-      groups.push(`${label}-${unit}-${values[0]}-${values[1]}`);
+  // Retrieve parameters if exists
+  if (urlsSplitted !== undefined && urlsSplitted.length > 1) {
+    queryParams = getQueryParameters(urlsSplitted[1]);
+  }
+
+  // Check if q param is present, add it if missing
+  let found = false;
+  queryParams.forEach((query) => {
```
Comment (JBLach) [inline src/js/modules/facetedsearch/filter-handler.ts:27]:
```suggestion
  queryParams.forEach((query: any) => {
```

**Review by JBLach — COMMENTED** [review]
Hi @Hlavtox 
I added suggestions that should fix the problem :)

**Review by matthieu-rolland — APPROVED** [review]

**Comment by matthieu-rolland** [comment]
thank you @Hlavtox 👍

**Pull request merged**